internal/connector/internal/workers/namespace_mgr.go

```diff
@@ -41,72 +42,51 @@ func NewNamespaceManager(namespaceService services.ConnectorNamespaceService, db
 
 func (m *NamespaceManager) Reconcile() []error {
 
-	if err := m.initContext(); err != nil {
-		return []error{err}
-	}
-
 	var errs []error
-	glog.V(5).Infof("Deleting expired namespaces...")
-	namespaces, err := m.namespaceService.GetExpiredNamespaceIds()
-	if err != nil {
-		errs = append(errs, err)
-	} else {
-		n := len(namespaces)
-		if n == 0 {
-			glog.V(5).Infof("No expired namespaces")
-		} else {
-
-			glog.V(5).Infof("Deleting %d namespaces...", n)
-			success := n
-			if serr := InDBTransaction(m.ctx, func(ctx context.Context) error {
-				for _, id := range namespaces {
-					if err := m.namespaceService.Delete(ctx, id); err != nil {
-						errs = append(errs, err)
-						success--
-					}
-				}
-				return nil
-			}); serr != nil {
-				errs = append(errs, serr)
-			}
-			glog.V(5).Infof("Deleted %d expired namespaces with %d errors", success, len(errs))
+	if m.ctx == nil {
+		ctx, err := m.db.NewContext(context.Background())
+		if err != nil {
+			return []error{err}
 		}
+		m.ctx = ctx
 	}
 
```

Comment:
Was the case of a namespace in deleting but for which a connector was created and destroyed i.e.  a connector with `deletad_at` not null ?

Reply:
The case we were targeting was namespace with `deleting` but with connectors with `not null` and `desired_state!=deleted`. That's handled by `ReconcileUsedDeletingNamespaces()`.

Reply:
So the assumption is that namespaces that had connectors assigned but already deleted and are in deleting state don't need reconciliation since sooner or later the data plane will place them in deleted state and they will be delete? Even in the unlikely scenario in which they will have another connector created (say a timing issue while placing them in deleting state) the next reconcile should take care of them with `ReconcileUsedDeletingNamespaces`? 

It that correct? Sorry I am thinking the unthinkable, but :shrug:

Reply:
Yes to both questions.